apps/builder/app/shared/copy-paste/plugin-instance.ts

```diff
@@ -164,15 +265,25 @@ export const onPaste = (clipboardData: string) => {
       propsStore,
       styleSourceSelectionsStore,
       stylesStore,
+      assetsStore,
     ],
     (
       breakpoints,
       instances,
       styleSources,
       props,
       styleSourceSelections,
-      styles
+      styles,
+      assets
     ) => {
+      // if pasting to a different project, copy assets.
+      // use the same ids so the references are preserved.
+      if (data.projectId !== project.id) {
+        for (const asset of data.assets) {
+          assets.set(asset.id, { ...asset, projectId: project.id });
```

Comment:
Or we can always iterate and check if id exists already before writing

Reply:
Is there a reason to choose one over the other?

Reply:
Everything else works this way. And here we additionally need to store project id in clipboard.